source/docs/software/advanced-controls/state-space/state-space-pose_state-estimators.rst

```diff
@@ -0,0 +1,39 @@
+
+The pose estimators perform latency compensation by storing a list of past observer states, including estimated state :math:`\mathbf{\hat{x}}`, error covariance :math:`\mathbf{P}`, inputs and local measurements. When new measurements are applied, the state of the estimator is first rolled back to the measurement's timestamp. Then, the filter corrects its state estimate with the new measurement and applies the inputs between the measurement timestamp and the present time to incorporate the new measurement. This allows for vision solutions with framerates which might otherwise make them unusable be a viable solution for robot localization.
+
+The following example shows the use of the ``DifferentialDrivePoseEstimator``:
+
+.. tabs::
+
+   .. code-tab:: java
+
+      var estimator = new DifferentialDrivePoseEstimator(new Rotation2d(), new Pose2d(),
+              new MatBuilder<>(Nat.N5(), Nat.N1()).fill(0.02, 0.02, 0.01, 0.02, 0.02), // State measurement standard deviations. X, Y, theta.
+              new MatBuilder<>(Nat.N3(), Nat.N1()).fill(0.02, 0.02, 0.01), // Local measurement standard deviations. Left encoder, right encoder, gyro.
+              new MatBuilder<>(Nat.N3(), Nat.N1()).fill(0.1, 0.1, 0.01)); // Global measurement standard deviations. X, Y, and theta.
+
+   .. code-tab:: cpp
+
+      frc::DifferentialDrivePoseEstimator estimator{
+              frc::Rotation2d(), frc::Pose2d(),
+              frc::MakeMatrix<5, 1>(0.01, 0.01, 0.01, 0.01, 0.01),
+              frc::MakeMatrix<3, 1>(0.1, 0.1, 0.1),
+              frc::MakeMatrix<3, 1>(0.1, 0.1, 0.1)};
+
+Tuning Pose Estimators
+----------------------
+
+All pose estimators offer user-customizable standard deviations for model and measurements. These standard deviations determine how much the filter "trusts" each of these states. For example, increasing the standard deviation for measurements (as one might do for a noisy signal) would lead to the estimator trusting its state estimate more than the incoming measurements. On the field, this might mean that the filter can reject noisy vision data well, at the cost of being slow to correct for model deviations. While these values can be estimated beforehand, they very much depend on the unique setup of each robot and global measurement method.
+
+The Differential Drive State Estimator
+--------------------------------------
+
+The ``DifferentialDriveStateEstimator`` (C++/Java) is a more advanced estimator which includes u-error estimation and encoder distance measurements to more accurately estimate robot position. The ``DifferentialDriveStateEstimator`` uses a system with the states :math:`\begin{bmatrix} x & y & theta & vel_l & vel_r & dist_l & dist_r & voltError_l & voltError_r & headingError \end{bmatrix}^T`, outputs left and right voltages, and can take inputs from either vision (as :math:`\begin{bmatrix}x & y & \theta \end{bmatrix}^T`) or encoders and gyros (as  :math:`\begin{bmatrix}dist_l & dist_r & \theta \end{bmatrix}^T`).
```

Comment:
u-error estimation has never been mentioned before so new readers won't know why this is a good thing.